pkg/linter/linter.go

```diff
@@ -315,8 +332,21 @@ func (l Linter) Lint(ctx context.Context) (report.Report, error) {
 		}
 	}
 
-	if len(input.FileNames) > 1 {
-		aggregateReport, err := l.lintWithRegoAggregateRules(ctx, regoReport.Aggregates, regoReport.IgnoreDirectives)
+	if len(input.FileNames) > 1 || len(l.additionalAggregates) > 0 {
+		allAggregates := make(map[string][]report.Aggregate)
+		for k, aggregates := range l.additionalAggregates {
+			allAggregates[k] = append(allAggregates[k], aggregates...)
+		}
+
+		for k, aggregates := range goReport.Aggregates {
+			allAggregates[k] = append(allAggregates[k], aggregates...)
+		}
+
+		for k, aggregates := range regoReport.Aggregates {
+			allAggregates[k] = append(allAggregates[k], aggregates...)
+		}
```

Comment:
Does this merging need to be smarter? e.g. overwrite state from additional aggs if there are new ones?